velox/type/Type.h

```diff
@@ -977,7 +962,42 @@ using TimestampType = ScalarType<TypeKind::TIMESTAMP>;
 using VarcharType = ScalarType<TypeKind::VARCHAR>;
 using VarbinaryType = ScalarType<TypeKind::VARBINARY>;
 using DateType = ScalarType<TypeKind::DATE>;
-using IntervalDayTimeType = ScalarType<TypeKind::INTERVAL_DAY_TIME>;
+
+class IntervalDayTimeType : public BigintType {
```

Comment:
Shouldn't this be defined under `velox/functions/prestosql/types/`? Is there any reason why this needs to be a "Velox builtin type" and not a "Presto UDT"?

Reply:
This is a good question. Originally it was added as a built-in Velox type. In this PR, I wanted to convert it from physical type to logical type primarily to explore what would that entail. Do you think I should make it Presto-specific type?

Reply:
I'm not sure I have enough context, but at a first thought, if it's type that follows Presto-specific semantic, should we try to keep it with the other Presto-specific types? Unless we think it's something that could have a high-degree of reuse across other dialects.

Reply:
I'm not sure either. DuckDB has this type and Spark seems to have something similar: https://spark.apache.org/docs/latest/sql-ref-datatypes.html